modules/core/05-port/keeper/keeper.go

```diff
@@ -87,5 +88,21 @@ func (k *Keeper) LookupModuleByPort(ctx context.Context, portID string) (string,
 // Route returns a IBCModule for a given module, and a boolean indicating
 // whether or not the route is present.
 func (k *Keeper) Route(module string) (types.IBCModule, bool) {
-	return k.Router.GetRoute(module)
+	routes, ok := k.Router.GetRoute(module)
+
+	if ok {
+		return routes, true
+	}
+
+	for _, prefix := range k.Keys() {
+		if strings.Contains(module, prefix) {
+			return k.Router.GetRoute(prefix)
+		}
+	}
+
+	return nil, false
+}
+
+func (k *Keeper) Keys() []string {
+	return k.Router.Keys()
```

Comment:
I think @chatton suggested before not having funcs which are one line. I think it'd make sense to just use this for the iteration above

Reply:
Good point! I started by copy-pasting from the feature branch :D but in that case I think the underlying router was not exported. No need for this method here!